### PR TITLE
fix: max image dimensions are now configurable and have a higher default

### LINKED ISSUE #41192: [QA] typical Photos from a smartphone are no longer rendered in the builtin image viewer
Seen in 10.14.0-rc.2

* Make a photo with a phone.
   - E.g. my old Samsung S3 from 2015 produced JPEGs with 9.6 MegaPixels (4128 x 2322)
   - more recent phones often have higher MegaPixel counts. 
* Upload such a photo to the Photos folder. Refresh. Wait. Refresh. No thumbnail icon appears. BAD.
* Click on the photo, files_mediaviewer opens. But nothing is shown. BAD
grafik

----

### Expected behaviour
* "normal" sized photos can be viewed in the media-viewer.

This behaviour apparently "works as designed" due to https://github.com/owncloud/core/blame/master/lib/private/Preview/Image.php#L49
Do we know, why this change was made? Such a hardcoded limit is disruptive from an end user perspective.

### Suggested behaviour
* Make the exact values configurable (as is, a change will trigger the integrity checker)
* make the default values an order of magnitude higher

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Related Issue
- Fixes #41192 

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: config/config.sample.php [hunk 1]
```diff
@@ -1020,6 +1020,14 @@
...
 'preview_max_filesize_image' => 50,
 
+/**
+ * Define the maximum image dimension for preview generation
+ *
+ * Value represents the maximum dimension in den format width x height
+ * Default is 6016x4000.
+ */
+'preview_max_dimensions' => '6016x4000',
+
```
Comment (jnweiger) [inline config/config.sample.php:1030]: A bit further up, we already have this:
```
/**
 * Define the maximum x-axis width for previews
 * The maximum width, in pixels, of a preview.
 * A value of `null` means there is no limit.
 */
'preview_max_x' => 2048,

/**
 * Define the maximum y-axis width for previews
 * The maximum height, in pixels, of a preview. A value of `null` means there is no limit.
 */
'preview_max_y' => 2048,
```
That is confusing now. Those values are used in apps/files_sharing/lib/Controllers/ShareController.php and lib/private/Preview.php

**Review by jnweiger — CHANGES_REQUESTED** [review]
We need to clarify the relation between
preview_max_dimensions and preview_max_x/preview_max_y

Sorry, I did not spot the latter pair earlier.

File: config/config.sample.php (hunk 1 above)
Comment (DeepDiver1975) [reply config/config.sample.php:1030]: These are output values.
And the new prop defined input values.

But I see your point...... Let me think....

File: config/config.sample.php [hunk 2]
```diff
@@ -1020,6 +1020,18 @@
  */
 'preview_max_filesize_image' => 50,
 
+/**
+ * Define the maximum dimensions of the original image for preview generation
+ * In contrary to `preview_max_x` and `preview_max_x` which define the maximum
```
Comment (phil-davis) [inline config/config.sample.php:1025]:
```suggestion
 * In contrast to `preview_max_x` and `preview_max_x` which define the maximum
```

File: config/config.sample.php [hunk 3]
```diff
@@ -1020,6 +1020,18 @@
...
+ * dimensions of generated previews, this setting limits the original images size.
```
[the remark this answers is not in the document]
Comment (DeepDiver1975) [reply config/config.sample.php:1026]: No idea TBH. Depends on the doc generation tool.... :shrug: 

@mmattel would know ...

**Review by phil-davis — APPROVED** [review]

**Review by jnweiger — APPROVED** [review]
Ah, thanks for clarification.

**Pull request merged**